cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/validator/AwsIDBrokerMappedRolePermissionValidator.java

```diff
@@ -73,20 +86,24 @@ public abstract class AwsIDBrokerMappedRolePermissionValidator extends AbstractA
      * @param cloudFileSystem         cloud file system to evaluate
      * @param validationResultBuilder builder for any errors encountered
      */
-    public void validate(AmazonIdentityManagementClient iam, CloudS3View cloudFileSystem,
-            ValidationResultBuilder validationResultBuilder) {
+    public void validate(AmazonIdentityManagementClient iam, CloudS3View cloudFileSystem, String backupLocation,
+            ValidationResultBuilder validationResultBuilder, InstanceProfile instanceProfile) {
         AccountMappingBase accountMappings = cloudFileSystem.getAccountMapping();
         if (accountMappings != null) {
             SortedSet<String> roleArns = getRoleArnsForUsers(getUsers(), accountMappings.getUserMappings());
             LOGGER.info("Getting role from AWS, roleArns.size: {}, roleArns: {}", roleArns.size(), roleArns);
             Set<Role> roles = awsIamService.getValidRoles(iam, roleArns, validationResultBuilder);
 
             boolean s3guardEnabled = cloudFileSystem.getS3GuardDynamoTableName() != null;
+            boolean validateBackupLocation = validateBackup(backupLocation);
```

Comment:
I would rename it(both the variable and method names) to refer that it actually marking the enablement of the validation it's self and not do the real validation here.

Reply:
we are checking two things
1. Check the entitlement
2. Check the backup location.